### PR TITLE
Fix described API in function comment

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: lib/proto_kvlds/proto_kvlds.h [hunk 1]
```diff
@@ -109,9 +109,10 @@ int proto_kvlds_request_get(struct wire_requestqueue *,
  * < ${end} via the request queue ${Q}.  Invoke
  *     ${callback}(${cookie}, failed, nkeys, next, keys, values)
  * upon request completion, where failed is 0 on success and 1 on failure,
- * nkeys is the number of key-value pairs returned, and keys and values are
- * arrays of keys and values respectively.  The callback is responsible for
- * freeing the provided structures.
+ * nkeys is the number of key-value pairs returned, next is the subsequent key
```
Comment (cperciva) [inline lib/proto_kvlds/proto_kvlds.h]: `next` isn't guaranteed to be the next key; rather, it's guaranteed to be a key such that issuing another request with `next` as the starting point will result in the subsequent key being the first one returned.

e.g., if there are keys "hello" and "world", we could return with `nkeys = 1` and a `next` value of "world", but it could equally well be "w", "m", or "hello world".  All we're guaranteed is that the last key returned < next <= the first key not returned.

**Comment by gperciva** [comment]
Um. That's a bit wordy for a clause in the middle of a sentence, but I guess it's better to be wordy and accurate, rather than inaccurate.

Updated PR.

**Comment by cperciva** [comment]
Yeah, I agree about the wordiness.  That might be why I left it out originally...
